### PR TITLE
Final pass through PropertyGrid

### PR DESCRIPTION
**Pull request by JeremyKuhne**
One last code cleanup pass for the PropertyGrid related code.

Trying to clear informational messages as part of the cleanup.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5312)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/MultiPropertyDescriptorGridEntry.cs [hunk 1]
```diff
@@ -153,20 +156,15 @@ protected override bool CreateChildren(bool diffOldChildren)
         }
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/MultiPropertyDescriptorGridEntry.cs:156]: Consider adding Debug.Fail on line 154 where you catch all exceptions.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyDescriptorGridEntry.cs [hunk 2]
```diff
@@ -396,21 +373,14 @@ protected object GetPropertyValueCore(object target)
                 target = descriptor.GetPropertyOwner(_propertyInfo);
             }
 
-            try
-            {
-                return _propertyInfo.GetValue(target);
-            }
-            catch
-            {
-                throw;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyDescriptorGridEntry.cs:405]: Nice one!

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewEdit.cs [hunk 3]
```diff
@@ -311,14 +258,9 @@ protected override bool ProcessCmdKey(ref Message msg, Keys keyData)
                 return base.ProcessCmdKey(ref msg, keyData);
             }
 
-            /// <summary>
-            ///  Overrides Control.ProcessDialogKey to handle the Escape and Return
-            ///  keys.
-            /// </summary>
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewEdit.cs:317]: Add <inheritdoc/>

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewListBoxItemAccessibleObject.cs [hunk 4]
```diff
@@ -70,17 +58,12 @@ internal override bool IsPatternSupported(UiaCore.UIA patternId)
             ///  Gets the runtime ID.
             /// </summary>
             internal override int[] RuntimeId
-            {
-                get
+                => new int[]
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewListBoxItemAccessibleObject.cs:61]: @vladimir-krestov @SergeySmirnov-Akvelon @DmitryGorokhov - please see this pattern, would be nice to use it as you go through the code!

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewListBoxItemAccessibleObject.cs [hunk 5]
```diff
@@ -70,17 +58,12 @@ internal override bool IsPatternSupported(UiaCore.UIA patternId)
             ///  Gets the runtime ID.
             /// </summary>
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewListBoxItemAccessibleObject.cs:59]: `<inheritdoc/>`?

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.MouseHook.MouseHookObject.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                    try
+                    {
+                        if (_reference.Target is MouseHook hook)
+                        {
+                            result = hook.MouseHookProc(nCode, wparam, lparam);
+                        }
+                    }
+                    catch
+                    {
+                        // ignore
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.MouseHook.MouseHookObject.cs:39]: Debug.Fail?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs [hunk 7]
```diff
@@ -343,12 +343,14 @@ public void PropertyGrid_BackColor_SetWithHandler_CallsBackColorChanged()
         {
             using var control = new PropertyGrid();
             int callCount = 0;
-            EventHandler handler = (sender, e) =>
+
+            void handler(object sender, EventArgs e)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs:347]: Is it possible to factor this helper out?

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good!

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridErrorDialog.cs [hunk 8]
```diff
@@ -0,0 +1,332 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridErrorDialog.cs:1]: Have you made any manual changes in this file?

What's we've traditionally been doing is that all type extractions and other code movements are done in own commit without any additional manual changes. Then all manual changes added in subsequent commits. This way it is obvious what manual changes are, and easier to track any regressions.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewEdit.cs [hunk 9]
```diff
@@ -351,9 +293,9 @@ protected override bool ProcessDialogKey(Keys keyData)
 
             protected override void SetVisibleCore(bool value)
             {
-                Debug.WriteLineIf(CompModSwitches.DebugGridView.TraceVerbose, "DropDownHolder:Visible(" + (value.ToString()) + ")");
-                // make sure we don't have the mouse captured if
-                // we're going invisible
+                Debug.WriteLineIf(CompModSwitches.DebugGridView.TraceVerbose, $"DropDownHolder:Visible({value})");
+
+                // make sure we don't have the mouse captured if we're going invisible.
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewEdit.cs:298]:
```suggestion
                // Make sure we don't have the mouse captured if we're going invisible.
```

**Review by RussKie — COMMENTED** [review]
👍

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridErrorDialog.cs (hunk 8 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridErrorDialog.cs:1]: I renamed the file.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs (hunk 7 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridTests.cs:347]: Yeah, I didn't bother as I was just running the auto-fix. I'll consider that when I write more tests in here.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.MouseHook.MouseHookObject.cs (hunk 6 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.MouseHook.MouseHookObject.cs:39]: Added. I'd love to dial back the aggressiveness more, but it is getting kind of late for 6.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewListBoxItemAccessibleObject.cs (hunk 5 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewListBoxItemAccessibleObject.cs:59]: Fixing this particular one, but there are plenty of these left in these classes.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewListBoxItemAccessibleObject.cs (hunk 4 above)
Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.GridViewListBoxItemAccessibleObject.cs:61]: Looks good

**Review by dreddy-work — APPROVED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**